docs/PFC-test-plan.md

```diff
@@ -0,0 +1,200 @@
+    - [Test Case #1 - PFC PAUSE with single lossless priority](#test-case-1---pfc-pause-with-single-lossless-priority)
+      - [Test Objective](#test-objective)
+      - [Test Configuration](#test-configuration)
+      - [Test Steps](#test-steps)
+    - [Test Case #2 - PFC PAUSE with multiple lossless priorities](#test-case-2---pfc-pause-with-multiple-lossless-priorities)
+      - [Test Objective](#test-objective-1)
+      - [Test Configuration](#test-configuration-1)
+      - [Test Steps](#test-steps-1)
+    - [Test Case #3 - PFC PAUSE with lossy priorities](#test-case-3---pfc-pause-with-lossy-priorities)
+      - [Test Objective](#test-objective-2)
+      - [Test Configuration](#test-configuration-2)
+      - [Test Steps](#test-steps-2)
+
+## Overview
+
+Traditional IEEE 802.3 Ethernet defines an unreliable communication
+medium. In a network path that normally consists of multiple hops
+between a source and destination, lack of feedback between transmitters
+and receivers at each hop is one of the main reasons that causes network
+unreliability. Transmitters can send packets faster than receivers can
+accept packets. As a result the receivers\' buffer keeps growing and
+beyond a threshold it starts silently dropping incoming packets. Flow
+control functions at Layer 2 offers a solution to this problem. Flow
+control enables feedback from a receiver to its sender to communicate
+buffer availability. IEEE 802.3x PAUSE control frames are introduced to
+achieve Layer 2 flow control. A receiver can send a PAUSE request to a
+sender when it senses potential buffer overflow. Upon receiving it, the
+sender stops transmission of any new packets on the wire until the
+receiver is ready to accept them again.
+
+IEEE 802.3x PAUSE works as designed but imposes certain limitations.
+Once a link is paused, the sender cannot send any traffic on that. As a
+result, an Ethernet Segment that carries multiple application flows with
+different QoS requirement, gets entirely blocked irrespective of their
+QoS. Thus enabling PAUSE for one application can affect the performance
+of the other network applications sharing the same Ethernet segment.
+IEEE 802.1Qbb PFC (Priority Flow Control) extends the basic PAUSE
+mechanism for multiple CoS, thus enabling coexistence of flows which
+needs flow control with other flows which performs better without it.
+Upto 8 CoSs can be used. PFC uses the same 64Byte MAC Control frame
+format that PAUSE frames do. It has a two byte Class-Enable vector to
+mention the CoS value for which the PAUSE should apply. As PFC acts
+independently on eight different CoSs, the frame describes the PAUSE
+duration for each CoS. The pause duration for each priority is a 2-byte
+value that expresses time as a number of quanta, where each quanta
+represents the time needed to transmit 512 bits at the current network
+speed. For user data traffic, the CoS maps to either CoS values defined
+in 802.1Q VLAN tag or the IP DSCP values.
```

Comment:
1. Mention that PFC is widely used to enable RoCE deployments. 
2. Add a PFC packet figure.

Reply:
Taken care of.